editor/editor_file_dialog.cpp

```diff
@@ -1294,6 +1294,18 @@ void EditorFileDialog::_update_favorites() {
 	favorite->set_pressed(false);
 
 	Vector<String> favorited = EditorSettings::get_singleton()->get_favorites();
+
+	bool fav_changed = false;
+	for (String &fav_file : favorited) {
+		if (!dir_access->dir_exists(fav_file)) {
+			favorited.erase(fav_file);
```

Comment:
Is it safe to use `erase()` inside the loop? Or will it skip the next item in the array in case if `erase()` moves the following items forward?

Reply:
Now that you are talking about it, it seems like it could make problems. (if the godot Vector doesn't work differently). I am going to iterate over `size()` of the vector and increase iterator, if `erase()` occured. Good catch, thank you.

Reply:
You can just to a reverse for loop. For loop that starts at the array length and decrements the index variable.